cypress/integration/plugins/security-dashboards-plugin/tenancy_change_on_shortlink.js

```diff
@@ -0,0 +1,134 @@
+
+      cy.createIndexPattern(
+        'index-pattern1',
+        {
+          title: 's*',
+          timeFieldName: 'timestamp',
+        },
+        indexPatternGlobalTenantHeaderSetUp
+      );
+
+      cy.createIndexPattern(
+        'index-pattern2',
+        {
+          title: 'se*',
+          timeFieldName: 'timestamp',
+        },
+        indexPatternPrivateTenantHeaderSetUp
+      );
+    });
+
+    it('Tests that when the short URL is copied and pasted, it will route correctly with the right tenant', function () {
+      const randomNumber = Cypress._.random(0, 1e6);
+      const dashboardName = 'Cypress dashboard - ' + randomNumber;
+      // We are programmatically creating a dashboard so that the test
+      // always have the same view. An empty list would show the empty prompt.
+      // Also, this saves us some typing, clicking and waiting in the test.
+      cy.createDashboard(
+        {
+          title: dashboardName,
+        },
+        {
+          security_tenant: 'private',
+        }
+      );
+
+      // When creating the shortUrl, we don't want to have the security_tenant
+      // parameter in the url - otherwise it will be stored in the shortUrl
+      // itself, which would make this test obsolete.
+      // But it is also hard to get the tests running reliably when opening
+      // Dashboards without the parameter (tenant selector popup etc.).
+      // Hence, we do some navigation to "lose" the query parameter.
+      CURRENT_TENANT.newTenant = 'private';
+      cy.visit('/app/home', {
+        waitForGetTenant: true,
+        onBeforeLoad(window) {
+          // set up session storage as we would expect to emulate browser
+          window.sessionStorage.setItem(
+            'opendistro::security::tenant::show_popup',
+            false
+          );
+
+          window.localStorage.setItem(
+            'opendistro::security::tenant::saved',
+            '__user__'
+          );
+        },
+      });
+      // Navigate to the Dashboards app
+      cy.getElementByTestId('toggleNavButton').should('be.visible').click();
+      // After clicking the navigation, the security_tenant parameter should be gone
+      cy.get('[href$="/app/dashboards#/list"]').should('be.visible').click();
+
+      // The test subj seems to replace spaces with a dash, so we convert the dashboard name here too.
+      // Go to the dashboard we have created
+      const selectorDashboardName = dashboardName.split(' ').join('-');
+      cy.getElementByTestId(
+        'dashboardListingTitleLink-' + selectorDashboardName
+      )
+        .should('be.visible')
+        .click();
+
+      cy.getElementByTestId('savedObjectTitle').type(dashboardName);
+
+      cy.intercept({
+        method: 'POST',
+        url: '/api/saved_objects/_bulk_get',
+      }).as('waitForReloadingDashboard');
+      cy.getElementByTestId('confirmSaveSavedObjectButton').click();
+      cy.wait('@waitForReloadingDashboard');
+      cy.wait(2000);
```

Comment:
nit: If you are already waiting for the request to complete, why the additional 2s?